backend/v3/domain/organization.go

```diff
@@ -0,0 +1,302 @@
+package domain
+
+import (
+	"context"
+	"time"
+
+	"github.com/zitadel/zitadel/backend/v3/storage/database"
+)
+
+type OrgState string
+
+const (
+	OrgStateActive   OrgState = "active"
+	OrgStateInactive OrgState = "inactive"
+)
+
+type Organization struct {
+	ID         string    `json:"id,omitempty" db:"id"`
+	Name       string    `json:"name,omitempty" db:"name"`
+	InstanceID string    `json:"instanceId,omitempty" db:"instance_id"`
+	State      OrgState  `json:"state,omitempty" db:"state"`
+	CreatedAt  time.Time `json:"createdAt,omitzero" db:"created_at"`
+	UpdatedAt  time.Time `json:"updatedAt,omitzero" db:"updated_at"`
+
+	Domains []*OrganizationDomain `json:"domains,omitempty" db:"-"`
+}
+
+var _ Commander = (*CreateOrganizationCommand)(nil)
+
+type CreateOrganizationCommand struct {
+	InstanceID string `json:"instanceId"`
+	// ID is optional, if not set a new ID will be generated.
+	// It can be set using the [WithOrganizationID] option in [NewCreateOrganizationCommand].
+	ID   string `json:"id,omitempty"`
+	Name string `json:"name"`
+
+	// CreatedAt MUST NOT be set by the caller.
+	CreatedAt time.Time `json:"createdAt,omitzero"`
+
+	// Admins represent the commands to create the administrators.
+	// The Commanders MUST either be [AddOrgMemberCommand] or [CreateOrgMemberCommand].
+	Admins []Commander `json:"admins,omitempty"`
+}
+
+type CreateOrganizationCommandOpts interface {
+	applyOnCreateOrganizationCommand(cmd *CreateOrganizationCommand)
+}
+
+func NewCreateOrganizationCommand(instanceID, name string, opts ...CreateOrganizationCommandOpts) *CreateOrganizationCommand {
+	cmd := &CreateOrganizationCommand{
+		InstanceID: instanceID,
+		Name:       name,
+	}
+	for _, opt := range opts {
+		opt.applyOnCreateOrganizationCommand(cmd)
+	}
+	return cmd
+}
+
+// Execute implements [Commander].
+//
+// DISCUSS(adlerhurst): As we need to do validation to make sure a command contains all the data required
+// we can consider the following options:
+// 1. Validate the command before executing it, which is what we do here.
+// 2. Create an invoker which checks if the struct has a `Validate() error` method and call it in the chain of invokers.
+// While the the first one is more straightforward it bloats the execute method with validation logic.
+// The second one would allow us to keep the execute method clean, but could be more error prone if the method gets missed during implementation.
+func (cmd *CreateOrganizationCommand) Execute(ctx context.Context, opts *CommandOpts) (err error) {
```

Comment:
Isn't it missing the part where you create the events and populate them with the right payload?